packages/component_code_gen/templates/generate_apps.py

```diff
@@ -0,0 +1,466 @@
+      });
+    },
+    async getRaindrops($, collectionId, params) {
+      return this._makeRequest($, {
+        path: `/raindrops/${collectionId}`,
+        params,
+      });
+    },
+  },
+};
+```
+
+This object contains a `propDefinitions` property, which contains the definitions for the props of the app.
+
+The propDefinitions object contains two props: collectionId and raindropId. The collectionId prop is a string prop. The raindropId prop is also a string prop. The propDefinitions object also contains an `options` method. The `options` method is an optional method that can be defined on a prop. It is used to dynamically generate the options for a prop and can return a static array of options or a Promise that resolves to an array of options.
+
+This object contains a `props` property, which defines a single prop of type "app":
+
+```javascript
+import { axios } from "@pipedream/platform";
+export default {
+  type: "app",
+  app: "the_app_name",
+  propDefinitions: {
+    prop_key: {
+      type: "string",
+      label: "Prop Label",
+      description: "A description of the prop",
+      async options() {
+        const static_options = ["option 1", "option 2"]; // a static array of options
+        const dynamic_options = await this.getOptions(); // a Promise that resolves to an array of options.
+        return dynamic_options; // return the options
+      },
+    }
+  },
+  methods: {
+    async _makeRequest($ = this, opts) {
+      const {
+        method = "get",
+        path,
+        data,
+        params,
+        ...otherOpts
+      } = opts;
+      return await axios($, {
+        ...otherOpts,
+        method,
+        url: `https://api.the_app_name.com${path}`, // the base URL of the app API
+        headers: {
+          ...opts.headers,
+          "Authorization": `Bearer ${this.$auth.oauth_access_token}`, // the authentication type depends on the app
+        },
+        params,
+        data,
+      })
+    },
+    async getOptions() {
+      // the code to get the options
+      return await this._makeRequest({
+        ...opts,
+      })
+    },
+  },
+}
+```
+
+This lets the user connect their app account to the step, authorizing requests to the app API.
+
+`this` exposes the user's app credentials in the object `this.$auth`. For integrations where users provide static API keys / tokens, the $auth object contains properties for each key / token the user enters. For OAuth integrations, this object exposes the OAuth access token in the oauth_access_token property of the $auth object.
+
+The app can be a key-based app. For integrations where users provide static API keys / tokens, `this.$auth` contains properties for each key / token the user enters. Users are asked to enter custom fields. They are each exposed as properties in the object `this.$auth`. When you make the API request, use the format from the app docs. Different apps pass credentials in different places in the HTTP request, e.g. headers, url params, etc.
```

Comment:
Can you include this language dynamically based on the details of the app? i.e. we should know if it's an OAuth vs. key-based app, and we should know the list of available custom fields / properties of the `this.$auth` object, so we should probably just generate that text dynamically.